pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -78,12 +78,16 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 
 	templateData.ProvisioningIP = config.BootstrapProvisioningIP
 	templateData.ProvisioningNetwork = string(config.ProvisioningNetwork)
-	templateData.BaremetalEndpointOverride = fmt.Sprintf("http://%s/v1", net.JoinHostPort(config.APIVIP, "6385"))
-	templateData.BaremetalIntrospectionEndpointOverride = fmt.Sprintf("http://%s/v1", net.JoinHostPort(config.APIVIP, "5050"))
 	templateData.ExternalStaticIP = config.BootstrapExternalStaticIP
 	templateData.ExternalStaticGateway = config.BootstrapExternalStaticGateway
 	templateData.ExternalMACAddress = config.ExternalMACAddress
 
+	if len(config.APIVIPs) > 0 {
+		templateData.APIVIP = config.APIVIPs[0]
```

Comment:
At another place in the code, we make sure that we have an IPv4 on index 0 in case of dual-stack: https://github.com/openshift/installer/blob/e7c39f77ce018823a05caf962dc94493aa11ad87/pkg/types/validation/installconfig.go#L539

Reply:
Note that right now we only support ipv4-primary for dual stack, so this is a reasonable assumption. There will be other changes needed to support ipv6-primary (which there are currently no plans to do) so if/when we do that this can be updated too.

Reply:
Agreed. And I think it is OK for ensureIPv4IsFirstInDualStackSlice() to be called within the validation code.